app/scripts/services/editorial-comments.js

```diff
@@ -0,0 +1,584 @@
+            * @property sendingReply
+            * @type Boolean
+            * @default false
+            */
+            comment.sendingReply = false;
+
+            /**
+            * A flag indicating whether or not a comment is currently being
+            * resolved.
+            *
+            * @property resolving
+            * @type Boolean
+            * @default false
+            */
+            comment.resolving = false;
+
+            /**
+            * Sets comment's display status to collapsed.
+            * @method collapse
+            */
+            comment.collapse = function () {
+                this.showStatus = 'collapsed';
+                this.isReplyMode = false;
+            };
+
+            /**
+            * Sets comment's display status to expanded.
+            * @method expand
+            */
+            comment.expand = function () {
+                this.showStatus = 'expanded';
+            };
+
+            /**
+            * Changes comment's display status from expanded to collapsed or
+            * vice versa.
+            * @method toggle
+            */
+            comment.toggle = function () {
+                if ('expanded' === this.showStatus) {
+                    this.collapse();
+                } else {
+                    this.expand();
+                }
+            };
+
+            /**
+            * Puts comment into edit mode.
+            * @method edit
+            */
+            comment.edit = function () {
+                this.editing.comment = this.comment;
+                this.isEdited = true;
+                this.isReplyMode = false;
+            };
+
+            /**
+            * End comment's edit mode.
+            * @method cancel
+            */
+            comment.cancel = function () {
+                this.isEdited = false;
+            };
+
+            /**
+            * Asynchronously saves/updates the comment and ends the edit mode.
+            * @method save
+            */
+            comment.save = function () {
+                var comment = this,
+                    deferred = $q.defer(),
+                    url;
+
+                url = Routing.generate(
+                    'newscoop_gimme_articles_edit_editorial_comment',
+                    {
+                        number: article.articleId,
+                        language: article.languageData.id,
+                        commentId: comment.id
+                    },
+                    true
+                );
+
+                $http.post(
+                    url,
+                    {editorial_comment: comment.editing},
+                    {transformRequest: transform.formEncode}
+                ).success(function () {
+                    deferred.resolve();
+                    comment.comment = comment.editing.comment;
+                    comment.isEdited = false;
+                }).error(function () {
+                    deferred.reject();
+                });
```

Comment:
`success` and `error` handlers have been deprecated in Angular 1.4, I suggest gradually moving towards the normal `then()` method of the promises.